internal/rc/rc.go

```diff
@@ -0,0 +1,30 @@
+package rc
+
+import "sync/atomic"
+
+type Releaser struct {
+	release  func()
+	refcount int32
+}
+
+func NewReleaser(init int32, release func()) *Releaser {
+	return &Releaser{
+		release:  release,
+		refcount: init,
+	}
+}
+
+func (rc *Releaser) Decr() {
+	newCount := atomic.AddInt32(&rc.refcount, -1)
+	if newCount == 0 {
+		rc.release()
+		rc.release = nil
+	}
```

Comment:
Consider adding a test for a negative `newCount` and panicking if we over-release.

Reply:
Done.